### ISSUE TITLE
Error "Cannot load such file -- aws-sdk-ecrpublic (LoadError)" while trying to run Inspec profile against AWS resource pack

### ISSUE THREAD

**Issue by Aniket281292** [inspec/inspec#5494]
Error "cannot load such file -- aws-sdk-ecrpublic (LoadError)" encountered while trying to run an inspec profile to test against aws resources.
# Describe the problem

I have a inspec profile called as "aws-test" and couple of inspec tests(ekstest.rb abctest.rb) in controls folder. When i try to run those inspec tests by mentioning the test names in the command, i am able to execute these tests successfully and dont see any error. Commands used to execute these tests:
inspec exec ~/.../aws-test/controls/ekstest.rb -t aws://us-east-1
inspec exec ~/.../aws-test/controls/abctest.rb -t aws://us-east-1.

However when i am trying to run inspec profile by mentioning the profile name, i am seeing the following error:
**command**: inspec exec aws-test -t aws://us-east-1

Error: 
1: from /opt/inspec/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:83:in `require'
/opt/inspec/embedded/lib/ruby/2.7.0/rubygems/core_ext/kernel_require.rb:83:in `require': cannot load such file -- aws-sdk-ecrpublic (LoadError)

**Comment by r-goto** [comment]
Had the same issue but the following inspec.yml fixed it in my case. 

```
---
name: my-profile
title: AWS InSpec Profile
maintainer: The Authors
copyright: The Authors
copyright_email: you@example.com
license: Apache-2.0
summary: An InSpec Compliance Profile For AWS
version: 0.1.0
inspec_version: '>= 4.6.9'
depends:
  - name: inspec-aws
    url: https://github.com/inspec/inspec-aws/archive/1.0.1-for-inspec-4.7.tar.gz
supports:
  - platform: aws
```

`url: https://github.com/inspec/inspec-aws/archive/1.0.1-for-inspec-4.7.tar.gz` did it, I believe.

**Comment by Aniket281292** [comment]
@r-goto thanks for the suggestion, However i am still seeing the issue after changing the depends url to what you have mentioned in the comment

**Comment by Aniket281292** [comment]
@clintoncwolfe , any idea what could be the issue here as i have also tried updating the archive file looking at resolutions for similar issue. The concern being that inspec tests are run successfully when command with the test names (abc.rb) are used but failing when profile-name is used..

**Comment by clintoncwolfe** [comment]
I'm not sure about the issue with running control files vs running entire profiles, but I can say that aws-sdk-ecrpublic is a very recent addition to train-aws (the support library that pulls in the AWS SDK gems). It is included in train-aws-0.2.0 . Check to see what version you have in `/opt//inspec/embedded/lib/ruby/gems/2.7.0/gems/train-aws-*` . If it is less than 0.2.0, upgrade inspec to 4.36.4, the most recent version.

**Comment by Aniket281292** [comment]
@clintoncwolfe , thank you. The train-aws version was less than 0.2.0. Upgrading inspec worked for me. I dont see that issue anymore.
